components/BottomNavigation/src/MDCBottomNavigationView.m

```diff
@@ -0,0 +1,258 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomNavigationView.h"
+
+#import "MaterialMath.h"
+#import "MaterialRTL.h"
+#import "private/MDCBottomNavigationCell.h"
+
+static NSString *const kMDCBottomNavigationViewBadgeColorString = @"badgeColor";
+static NSString *const kMDCBottomNavigationViewBadgeValueString = @"badgeValue";
+static NSString *const kMDCBottomNavigationViewImageString = @"image";
+static NSString *const kMDCBottomNavigationViewTitleString = @"title";
+static NSString *const kMDCBottomNavigationViewNewString = @"new";
+
+@interface MDCBottomNavigationView ()
+
+@property(nonatomic, strong) NSMutableArray<MDCBottomNavigationCell *> *navBarCells;
+@property(nonatomic, assign) UIUserInterfaceLayoutDirection layoutDirection;
```

Comment:
I don't know that I would cache this, I would call self.mdc_effectiveUserInterfaceLayoutDirection where appropriate.